### DISCUSSION TITLE
useLinkProps breaking changes in v7

### DISCUSSION THREAD

**Discussion by CorentinAndre** [react-navigation/react-navigation#11760] [Q&A]
Following this https://reactnavigation.org/docs/7.x/upgrading-from-6.x#the-link-component-and-uselinkprops-hook-now-use-screen-names-instead-of-paths, I wanted to express my concerns over these changes.

Will we be able to still navigate via a link ? I'm asking this because since v5, we switched our codebase to navigating with links. I wrote some types which allow me to do something like this

`<Link to="SomeScreen"  params={{...}} />` where the params are typed depending on the path specified for SomeScreen.
It's easier for me to navigate to an URL than to a screen inside a navigator because screens & navigators structure can change without developers knowing (imagine someone adds a new navigator). This would break the nesting & thus the links inside our codebase. Whereas navigating to a link will always work as long as it exists in the codebase.

- Can someone explain me how navigating to a screen is "easier" to type ? 
- Does the option to navigate to a screen with an href will still exist in v7 ? 
It could be great to have both options (navigating with screens & links)

Thanks :)

**Comment by satya164** [comment]
> Can someone explain me how navigating to a screen is "easier" to type

There are no types for the path string. It's not "easier" to type, it's the **only way** where React Navigation provides types.

> Does the option to navigate to a screen with an href will still exist in v7

React Navigation doesn't use links internally. When you navigate via link, it gets parsed to an action that's used for navigation. Nothing prevents you from doing the same thing. The `Link` component and `useLinkProps` won't support it, however, your own hook/component can still do that if you want. There'll be helpers exported to parse a link to an action with the `useLinkBuilder` hook.
